client/signup/steps/user/index.jsx

```diff
@@ -361,7 +361,12 @@ export class UserStep extends Component {
 		const stepAfterRedirect =
 			getNextStepName( this.props.flowName, this.props.stepName ) ||
 			getPreviousStepName( this.props.flowName, this.props.stepName );
-		return this.originUrl() + getStepUrl( this.props.flowName, stepAfterRedirect );
+		const queryArgs = new URLSearchParams( this.props?.initialContext?.query );
```

Comment:
Pinging @Automattic/martech just to make sure this gets eyes from their side. I _think_ this should be OK, but it feels central enough to me that I am almost certainly missing possible downstream side-effects.

Reply:
Thanks for the ping. Is there a P2 or GH issue to get more context on the issue? From the testing instructions:

> From /start/business?coupon={one_of_the_coupons} you should only see the selected TLDs in the search results (in the sandboxed environment you'll see only one TLD)

Where can I find a coupon to test this?

Reply:
For more context (and the actual coupon codes) - pbBQWj-rA-p2#comment-1615

The idea is that we need the `coupon` query param to change how the flow works (suggest only certain TLDs and also hide the free wpcom subdomain). However if you're a new user the coupon gets lost in the user step so that's why I made the change to pass through all params

Reply:
Thanks for the context. I tested and LGTM.